src/common/transformations/src/transformations/common_optimizations/sdpa_scale_fusion.cpp

```diff
@@ -83,14 +83,16 @@ SDPAScaleFusion::SDPAScaleFusion() {
         // Extract scalar scale values for Q and K if those are constant and set new inputs for SDPA
         if (has_q_scale) {
             scale_q_node = pattern_map.at(scale_q).get_node_shared_ptr();
-            if (ov::is_type<ov::op::v0::Constant>(scale_q_node)) {
+            if (ov::is_type<ov::op::v0::Constant>(scale_q_node) &&
+                pattern_map.at(q).get_element_type() == q_input.get_element_type()) {
                 scale_q_value = ov::as_type_ptr<ov::op::v0::Constant>(scale_q_node)->cast_vector<float>()[0];
                 q_input = pattern_map.at(q);
             }
```

Comment:
```suggestion
            } else {
                has_q_scale = false;
            }
```

Reply:
Could you consider my suggestions?
If `q_input` is not updated, I think it should be treated as if `has_q_scale` is false.